regression/cpp/gcc_vector1/main.cpp

```diff
@@ -0,0 +1,22 @@
+#ifdef __GNUC__
+
+typedef float __m128 __attribute__ ((__vector_size__ (16), __may_alias__));
+
+__m128 setzero(void)
+{
+  return (__m128){ 0.0f, 0.0f, 0.0f, 0.0f };
+}
+
+#else
+
+void setzero()
```

Comment:
How come we're only testing on GNUC?

Reply:
I have no idea whether Visual Studio supports (the same) vector types?

Reply:
MSVC doesn't support the vector_size attribute. It does have [128-bit data types](https://docs.microsoft.com/en-us/cpp/cpp/m128). However, these are defined in header files, and the cbmc C++ front end cannot parse MSVC C++ header files.

Reply:
Is the syntax to use them the same as with GCC/Clang? Would you actually be able to contribute a `KNOWNBUG` regression test that uses them?

Reply:
The existing testcase uses the typedef for __m128 from the xmmintrin.h header files that ships with gcc. I can do the same for MSVC and add that to the existing test case in a separate PR (after this one is merged). There is an interesting problem, though: CBMC with `PARSER.mode==configt::ansi_ct::flavourt::VISUAL_STUDIO` accepts the syntax of the test case, but the MSVC compiler does not: it complains that ```error C4576: a parenthesized type followed by an initializer list is a non-standard explicit type conversion syntax```
Should cbmc in this case follow MSVC and report a parser error?

Reply:
It would be desirable if the CBMC front-end mimicked exactly what MSVC does, but that's surely not of the highest priority.